### DISCUSSION TITLE
How do I get updates for debian docker with custom names ?

### DISCUSSION THREAD

**Discussion by azro352** [renovatebot/renovate#25038] [Request Help] [answered]
> Self-hosted v37.2.0-full // GitLab self-hosted

I never saw this working and I want the suggested update from docker  `debian-custom:stretch` to `debian-custom:buster`

### What you tried so far.

I have a `debian` docker image but customized, so it isn't named `/debian:stretch` but `/debian-custom:stretch` and I want to update the debian version

I've try adding a custom regex to match the full tag as a `major`, I've tried adding the preset `docker:enableMajor`, but none works, I don't receive any updates

Note that in the logs I see the 200 http code on the `custom.apifactory.url/.../debian-custom/tags/list` that lists the available versions and it well contains a `buster` and `bullseye` images

```js
module.exports = {
    packageRules: [
        { // for version without patch, only 1.1-buster, like python images
            matchDatasources: ["docker"],
            versioning: "regex:^(?<major>\\d+)\\.(?<minor>\\d+)(?<prerelease>[^.-]+)?(-(?<build>.*))?$"
        },
        { // for debian version only, like buster
            matchDatasources: ["docker"],
            versioning: "regex:^(?<major>.*)$"
        },
    ],
    // CI is splitted in files in the .ci directory
    gitlabci: {
        fileMatch: [
            "\\.ci/.*\\.ya?ml$"
        ]
    },
    extends: [
        "docker:enableMajor"
    ]
}
```

### Relevant debug logs

```
 "gitlabci": [
   {
     "packageFile": ".ci/build.yml",
     "deps": [
       {
         "depName": "custom.apifactory.url/docker/debian-custom",
         "currentValue": "stretch",
         "replaceString": "custom.apifactory.url/docker/debian-custom:stretch",
         "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
         "datasource": "docker",
         "depType": "image",
         "updates": [],
         "packageName": "custom.apifactory.url/docker/debian-custom",
         "versioning": "regex:^(?<major>.*)$",
         "warnings": [],
         "registryUrl": "https://custom.apifactory.url",
         "currentVersion": "stretch",
         "fixedVersion": "stretch"
       }
     ]
   }
 ]
```

**Comment by rarkins** [comment]
Capture the whole tag as currentValue, set versioning=debian

**Reply by azro352** [reply, marked as answer]
Am I not already capturing the whole tag as `currentValue` ? also if I set `versionning: debian` I can't keep the versionning with my regex

May i replace with `matchPackagePatterns` to filter, then specify the debian versionning system ? 

```js
{
  "packageRules": [
    {
      matchDatasources: ["docker"],
      matchPackagePatterns: ["debian-custom"],
      versionning: "debian"
    }
  ]
}
```

**Reply by rarkins** [reply]
I don't think you have a need for regex manager or versioning
